p11mod/p11mod.go

```
@@ -410,9 +410,29 @@ func (ll *llBackend) CreateObject(sh pkcs11.SessionHandle, temp []*pkcs11.Attrib
 }
 
 func (ll *llBackend) CopyObject(sh pkcs11.SessionHandle, o pkcs11.ObjectHandle, temp []*pkcs11.Attribute) (pkcs11.ObjectHandle, error) {
-	// TODO
-	log.Println("p11mod CopyObject: not implemented")
-	return 0, pkcs11.Error(pkcs11.CKR_FUNCTION_NOT_SUPPORTED)
+	session, err := ll.getSessionByHandle(sh)
+	if err != nil {
+		return 0, err
+	}
+
+	objectIndex := int(o - 1)
+
+	if objectIndex < 0 || objectIndex >= len(session.objects) {
+		log.Printf("p11mod CopyObject: index invalid: requested %d, object count %d\n", objectIndex, len(session.objects))
```

Please return an error in this case; logging is good but not sufficient.

---

return error added